internal/shim-sgx/src/handler/mod.rs

```diff
@@ -142,15 +144,43 @@ impl guest::Handler for Handler<'_> {
         Ok(())
     }
 
-    // Until EDMM, we can't change any page permissions.
-    // What you get is what you get. Fake success.
     fn mprotect(
         &mut self,
         _platform: &impl Platform,
-        _addr: NonNull<c_void>,
-        _len: size_t,
-        _prot: c_int,
+        addr: NonNull<c_void>,
+        length: size_t,
+        prot: c_int,
     ) -> sallyport::Result<()> {
+        let user_mem_scope = UserMemScope;
+
+        unsafe {
+            self.syscall(
+                &user_mem_scope,
+                [
+                    libc::SYS_mprotect as usize,
+                    addr.as_ptr() as usize,
+                    length,
+                    prot as usize,
+                    0,
+                    0,
+                    0,
+                ],
+            )?;
+        };
```

Comment:
This should call ioctl directly https://github.com/enarx/sallyport/blob/d28863940e463ee32cb1e29d400db2d2300e3264/src/guest/handler.rs#L288-L300 instead

Otherwise, if that's not feasible, because host needs to hold a FD, for example, introduce a custom `enarxcall`, e.g. `RestrictSgxPermissions`.

Although it seems to me that cleanest way to handle this would be to have a way to get a FD to the shim and then call ioctls on that FD

So maybe we could have `OpenSgxEnclave` enarxcall?